lang/csharp/src/apache/main/AvroDecimal.cs

```diff
@@ -0,0 +1,929 @@
+        }
+
+        public static bool operator ==(AvroDecimal left, AvroDecimal right)
+        {
+            return left.Equals(right);
+        }
+
+        public static bool operator !=(AvroDecimal left, AvroDecimal right)
+        {
+            return !left.Equals(right);
+        }
+
+        public static bool operator >(AvroDecimal left, AvroDecimal right)
+        {
+            return left.CompareTo(right) > 0;
+        }
+
+        public static bool operator >=(AvroDecimal left, AvroDecimal right)
+        {
+            return left.CompareTo(right) >= 0;
+        }
+
+        public static bool operator <(AvroDecimal left, AvroDecimal right)
+        {
+            return left.CompareTo(right) < 0;
+        }
+
+        public static bool operator <=(AvroDecimal left, AvroDecimal right)
+        {
+            return left.CompareTo(right) <= 0;
+        }
+
+        public static bool operator ==(AvroDecimal left, decimal right)
+        {
+            return left.Equals(right);
+        }
+
+        public static bool operator !=(AvroDecimal left, decimal right)
+        {
+            return !left.Equals(right);
+        }
+
+        public static bool operator >(AvroDecimal left, decimal right)
+        {
+            return left.CompareTo(right) > 0;
+        }
+
+        public static bool operator >=(AvroDecimal left, decimal right)
+        {
+            return left.CompareTo(right) >= 0;
+        }
+
+        public static bool operator <(AvroDecimal left, decimal right)
+        {
+            return left.CompareTo(right) < 0;
+        }
+
+        public static bool operator <=(AvroDecimal left, decimal right)
+        {
+            return left.CompareTo(right) <= 0;
+        }
+
+        public static bool operator ==(decimal left, AvroDecimal right)
+        {
+            return left.Equals(right);
+        }
+
+        public static bool operator !=(decimal left, AvroDecimal right)
+        {
+            return !left.Equals(right);
+        }
+
+        public static bool operator >(decimal left, AvroDecimal right)
+        {
+            return left.CompareTo(right) > 0;
+        }
+
+        public static bool operator >=(decimal left, AvroDecimal right)
+        {
+            return left.CompareTo(right) >= 0;
+        }
+
+        public static bool operator <(decimal left, AvroDecimal right)
+        {
+            return left.CompareTo(right) < 0;
+        }
+
+        public static bool operator <=(decimal left, AvroDecimal right)
+        {
+            return left.CompareTo(right) <= 0;
+        }
+
+        public static explicit operator byte(AvroDecimal value)
+        {
+            return ToByte(value);
+        }
+
+        /// <summary>
+        /// Creates a byte from a given <see cref="AvroDecimal"/>.
+        /// </summary>
+        /// <param name="value">The <see cref="AvroDecimal"/>.</param>
+        /// <returns>A byte.</returns>
+        public static byte ToByte(AvroDecimal value)
```

Comment:
why expose all these static methods (in addition to the cast operators). Is there any precedent in the std lib? Doesn't seem consistent with how other numeric types work afaict. It seems to me it would be better to consolidate all of these into just the cast operators.

Reply:
So these are the implementation of `IConvertible`. VB.NET doesn't invoke the conversion operators and instead relies on the type implementing this interface.